src/test/java/com/google/sps/TimeRangeTest.java

```diff
@@ -0,0 +1,148 @@
+public final class TimeRangeTest {
+  Instant TIME_430PM = Instant.parse("2020-07-06T17:00:10.324978Z");
+  Instant TIME_5PM = Instant.parse("2020-07-06T17:00:10.324978Z");
+  Instant TIME_530PM = Instant.parse("2020-07-06T17:30:10.324978Z");
+  Instant TIME_6PM = Instant.parse("2020-07-06T18:00:10.324978Z");
+  Instant TIME_630PM = Instant.parse("2020-07-06T18:30:10.324978Z");
+  Instant TIME_645PM = Instant.parse("2020-07-06T18:45:10.324978Z");
+  Instant TIME_7PM = Instant.parse("2020-07-06T19:00:10.324978Z");
+  Instant TIME_730PM = Instant.parse("2020-07-06T19:30:10.324978Z");
+  Instant TIME_745PM = Instant.parse("2020-07-06T19:45:10.324978Z");
+  Instant TIME_8PM = Instant.parse("2020-07-06T20:00:10.324978Z");
+  Instant TIME_830PM = Instant.parse("2020-07-06T20:30:10.324978Z");
+  Instant TIME_9PM = Instant.parse("2020-07-06T21:00:10.324978Z");
+  Instant TIME_930PM = Instant.parse("2020-07-06T21:30:10.324978Z");
+
+  // Tests whether or not two TimeRanges are the same
+  @Test
+  public void equality() {
+    Assert.assertEquals(
+        TimeRange.fromStartEnd(TIME_530PM, TIME_6PM), TimeRange.fromStartEnd(TIME_530PM, TIME_6PM));
+
+    Assert.assertNotEquals(
+        TimeRange.fromStartEnd(TIME_530PM, TIME_6PM),
+        TimeRange.fromStartEnd(TIME_530PM, TIME_630PM));
+  }
+
+  // Tests whether or not a timerange contains a certain instant.
+  @Test
+  public void containsPoint() {
+    TimeRange range = TimeRange.fromStartEnd(TIME_530PM, TIME_630PM);
+    Assert.assertFalse(range.contains(TIME_7PM));
+    Assert.assertTrue(range.contains(TIME_6PM));
+  }
+
+  // Tests whether or not a TimeRange contains another TimeRange
+  @Test
+  public void containsRange() {
+    TimeRange range = TimeRange.fromStartEnd(TIME_630PM, TIME_8PM);
+
+    // |---|   |--range--|
+    Assert.assertFalse(range.contains(TimeRange.fromStartEnd(TIME_430PM, TIME_5PM)));
+
+    //     |--range--|
+    // |---|
+    Assert.assertFalse(range.contains(TimeRange.fromStartEnd(TIME_6PM, TIME_630PM)));
+
+    //   |--range--|
+    // |---|
+    Assert.assertFalse(range.contains(TimeRange.fromStartEnd(TIME_6PM, TIME_645PM)));
+
+    // |--range--|
+    // |---|
+    Assert.assertTrue(range.contains(TimeRange.fromStartEnd(TIME_630PM, TIME_7PM)));
+
+    // |--range--|
+    //    |---|
+    Assert.assertTrue(range.contains(TimeRange.fromStartEnd(TIME_7PM, TIME_730PM)));
+
+    // |--range--|
+    //       |---|
+    Assert.assertTrue(range.contains(TimeRange.fromStartEnd(TIME_730PM, TIME_8PM)));
+
+    // |--range--|
+    //         |---|
+    Assert.assertFalse(range.contains(TimeRange.fromStartEnd(TIME_745PM, TIME_830PM)));
+
+    // |--range--|
+    //           |---|
+    Assert.assertFalse(range.contains(TimeRange.fromStartEnd(TIME_8PM, TIME_830PM)));
+
+    // |--range--| |---|
+    Assert.assertFalse(range.contains(TimeRange.fromStartEnd(TIME_9PM, TIME_930PM)));
+  }
+
+  // Tests whether or not two TimeRanges overlap
+  @Test
+  public void overlaps() {
```

Comment:
Nit: Might be clearer to split this into a bunch of smaller tests whose names describe the behavior under test.

One naming scheme you could follow there is 

"overlaps_nonOverlapping"

"overlaps_startAndEnd"

etc.